dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/GetMaskedQuoteIdByReservedOrderId.php

```diff
@@ -51,15 +48,21 @@ public function __construct(
      * Get masked quote id by reserved order id
      *
      * @param string $reservedOrderId
+     * @param bool $forceCollectTotal
      * @return string
      * @throws NoSuchEntityException
      */
-    public function execute(string $reservedOrderId): string
+    public function execute(string $reservedOrderId, bool $forceCollectTotal = false): string
     {
         $quote = $this->quoteFactory->create();
         $quote->setSharedStoreIds(['*']);
         $this->quoteResource->load($quote, $reservedOrderId, 'reserved_order_id');
 
+        // If dataprovider is used, we need to collect totals manually and save quote
+        if ($forceCollectTotal) {
+            $this->quoteResource->save($quote->collectTotals());
```

Comment:
I guess saving quote introduced in a method named "get" which will violates single responsibility.